src/pages/experiments/[experimentId]/data-processing/configure-embedding/components/EmbeddingPreview.jsx

```diff
@@ -136,23 +186,40 @@ const EmbeddingPreview = () => {
   };
 
   const renderPlot = () => {
-    if (error) {
+    if (embeddingMethod && embedding[embeddingMethod]?.error) {
       return (
         <PlatformError
-          description={error}
-          onClick={() => { }}
+          description={embedding[embeddingMethod]?.error}
+          onClick={() => dispatch(loadEmbedding(experimentId, embeddingMethod))}
         />
       );
     }
 
-    if (!config || (embeddingMethod && embedding[embeddingMethod]?.loading)) {
+    if (cellSets && cellSets.error) {
+      return (
+        <PlatformError
+          description={cellSets.error}
+          onClick={() => dispatch(loadCellSets(experimentId))}
+        />
+      );
+    }
+
+    if (embeddingMethod && embedding[embeddingMethod]?.loading) {
       return (
         <center>
           <Spin size='large' />
         </center>
       );
     }
 
+    if (selectedSpec === 'sample'
+      && plotSpec.data.find((d) => d.name === 'cellSets').values.length === 0) {
+      return (
+        <Empty description='Your project has only one sample.' />
+
+      );
+    }
```

Comment:
@aerlaut -- this change should be incorporated into your MR to ensure the user sees a default state when they have only one sample.